modules/solid_mechanics/src/materials/ConstitutiveModel.C

```diff
@@ -93,13 +95,19 @@ ConstitutiveModel::applyThermalStrain(unsigned qp,
                                       SymmTensor & strain_increment,
                                       SymmTensor & d_strain_dT)
 {
-  if (_has_temp && _t_step != 0)
+  if (_t_step >= 1)
+    _zeroth_step = false;
+
+  if (_t_step >= 2)
+    _first_step = false;
+
```

Comment:
Two Booleans < one unsigned int

Either way, I'm not sure I really care, if you want to store a local copy of the time step, that's fine as long as you keep updating it so it actually makes sense. This implementation is also fine.